### PR TITLE
Out of schedule email for areas.

### PR DESCRIPTION
**Pull request by r-xyz**
Extends functionality to accesses outside the allowed schedule for areas not requiring a reservation.

Relevant area -related changes are mirrored from [NEMO-CE Merge Request](https://gitlab.com/nemo-community/nemo-ce/-/merge_requests/31).

### CODE REVIEW

**Comment by r-xyz** [comment]
@rptmat57 This should replicate relevant changes to upstream NEMO; please have a double check I did not miss anything.

File: NEMO/views/timed_services.py [hunk 1]
```diff
@@ -409,31 +410,81 @@ def send_email_out_of_time_reservation_notification(request=None):
...
+                )
+                # find out if a reservation is starting right at the same time (in case of back to back reservations, in which case customer is good)
+                starting_reservations = Reservation.objects.filter(
+                    cancelled=False, missed=False, shortened=False, area=area, user=customer, start=threshold
+                )
+                if ending_reservations.exists() and not starting_reservations.exists():
+                    out_of_time_user_reservations.append(ending_reservations[0])
+
+        # Find all users logged in tools
+        usage_records: List[UsageEvent] = (
```
Comment (rptmat57) [inline NEMO/views/timed_services.py:443]: this whole section should be removed. there are no qualification levels or tool requiring reservation in NEMO

**Review by rptmat57 — CHANGES_REQUESTED** [review]

File: NEMO/views/timed_services.py (hunk 1 above)
Comment (r-xyz) [reply NEMO/views/timed_services.py]: that was a big overlook, sorry. I fixed it, thanks for spotting.

File: resources/emails/out_of_time_reservation_email.html [hunk 2]
```diff
@@ -11,18 +11,38 @@
...
+                        {% endif %}
+                        {{ reservation.area }} in the {{ facility.name }} ended at {{ reservation.end }}.
+                    </p>
                     <p>However, our records show that you are still logged in the {{ reservation.area }}.</p>
                     <p>If you simply forgot to log out, please contact a staff member</p>
                     <br>
                     <p>
-                        Please remember that the facility is a shared resource, and overstaying a reservation may inhibit the productivity of other facility users.
+                        {% if reservation.start %}
+                            Please remember that the {{ facility.name }} is a shared resource, and overstaying a reservation may inhibit the productivity of other facility users.
```
Comment (rptmat57) [inline resources/emails/out_of_time_reservation_email.html:42]:
```suggestion
                            Please remember that the {{ facility_name }} is a shared resource, and overstaying a reservation may inhibit the productivity of other facility users.
```

File: resources/emails/out_of_time_reservation_email.html [hunk 3]
```diff
@@ -11,18 +11,38 @@
...
                     <p>Dear {{ reservation.user.first_name }},</p>
-                    <p>Your reservation of the {{ reservation.area }} in the facility ended at {{ reservation.end }}.</p>
+                    <p>
+                        Your
+                        {% if reservation.start %}
+                            reservation of the
+                        {% else %}
+                            allowed schedule for the
+                        {% endif %}
+                        {{ reservation.area }} in the {{ facility.name }} ended at {{ reservation.end }}.
```
Comment (rptmat57) [inline resources/emails/out_of_time_reservation_email.html:35]:
```suggestion
                        {{ reservation.area }} in the {{ facility_name }} ended at {{ reservation.end }}.
```

**Pull request merged**